LayoutTests/fast/dom/gc-abort-controller.html

```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>Test AbortController memory leak with event listeners</title>
+    <script src="../../resources/js-test.js"></script>
+</head>
+<body>
+<script>
+description("Test AbortController memory leak with event listeners");
+jsTestIsAsync = true;
+
+if (!window.testRunner || !window.internals) {
+    testFailed("Requires testRunner and internals.");
+    finishJSTest();
+}
+
+function runTest() {
+    let weakRefs = [];
+    let controllers = [];
+    
+    // Create 500 AbortControllers with event listeners
+    for (let i = 0; i < 500; i++) {
+        let controller = new AbortController();
+        controller.signal.addEventListener('abort', () => {
+            console.log('Event listener for controller ' + i);
+        });
+        controllers.push(controller);
+        weakRefs.push(new WeakRef(controller.signal));
+    }
+    
+    controllers = null;
+
+    setTimeout(() => {
```

Comment:
This is still likely to be flaky sorry. You can always use an existing gc test as example but essentially, you'll need a setInterval() to keep trying to `gc()` & check until some objects are collected (or a timeout is reached, in which case the test failed).